tests/spec/s-function.js

```diff
@@ -1,7 +1,177 @@
 /* global describe, it, expect, beforeEach */
+var hasStrictMode = (function () {
+  'use strict';
+
+   return !this;
+}());
+
+var global = (function () {
+   return this;
+}());
 
 describe('Function', function () {
-    'use strict';
+    describe('#call()', function () {
+        var noop = function () {};
+        it('should not throw a error in each case', function () {
+            expect(function () {
+                noop.call();
+            }).not.toThrow();
+            expect(function () {
+                /*eslint no-useless-call: 0*/
+                noop.call(undefined);
+            }).not.toThrow();
+            expect(function () {
+                /*eslint no-useless-call: 0*/
+                noop.call(null);
+            }).not.toThrow();
+            expect(function () {
+                noop.call('a');
+            }).not.toThrow();
+            expect(function () {
+                noop.call(1);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(true);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(noop);
+            }).not.toThrow();
+            expect(function () {
+                noop.call(new Date());
+            }).not.toThrow();
+            expect(function () {
+                noop.call(/test/);
+            }).not.toThrow();
+        });
+        it('should not throw a error with additional args', function () {
```

Comment:
I threw in all the tests that I had. There are sure to be improvements. I'll try and improve them some.